idaes/generic_models/properties/core/pure/ConstantProperties.py

```diff
@@ -0,0 +1,133 @@
+
+        @staticmethod
+        def return_expression(b, cobj, T):
+            # Specific heat capacity
+            cp = (cobj.cp_mol_liq_comp_coeff)
+            units = b.params.get_metadata().derived_units
+            return pyunits.convert(cp, units["heat_capacity_mole"])
+
+
+    class enth_mol_liq_comp(object):
+
+        @staticmethod
+        def build_parameters(cobj):
+            # if not hasattr(cobj, "cp_mol_liq_comp_coeff"):
+            #     cp_mol_liq_comp.build_parameters(cobj)
+
+            if cobj.parent_block().config.include_enthalpy_of_formation:
+                units = cobj.parent_block().get_metadata().derived_units
+
+                cobj.enth_mol_form_liq_comp_ref = Var(
+                        doc="Liquid phase molar heat of formation @ Tref",
+                        units=units["energy_mole"])
+                set_param_from_config(cobj, param="enth_mol_form_liq_comp_ref")
+
+            cobj.enth_mol_form_liq_comp_coeff = Var(
+                doc="Liquid phase molar heat of formation",
+                units=pyunits.kJ/pyunits.mol)
+            set_param_from_config(cobj, param="enth_mol_form_liq_comp_coeff")
+
+        @staticmethod
+        def return_expression(b, cobj, T):
+            # Specific enthalpy
+            T = pyunits.convert(T, to_units=pyunits.K)
+            Tr = pyunits.convert(b.params.temperature_ref, to_units=pyunits.K)
+
+            units = b.params.get_metadata().derived_units
+
+            h_form = (cobj.enth_mol_form_liq_comp_ref if
+                    b.params.config.include_enthalpy_of_formation
+                    else 0*units["energy_mole"])
+
+            h = (pyunits.convert(
+                    (cobj.enth_mol_form_liq_comp_coeff), units["energy_mole"]) +
+                h_form)
+
+            return h
+
+
+    class entr_mol_liq_comp(object):
+
+        @staticmethod
+        def build_parameters(cobj):
+            # if not hasattr(cobj, "cp_mol_liq_comp_coeff_1"):
+            #     cp_mol_liq_comp.build_parameters(cobj)
+
+            units = cobj.parent_block().get_metadata().derived_units
+
+            cobj.entr_mol_form_liq_comp_ref = Var(
+                    doc="Liquid phase molar entropy of formation @ Tref",
+                    units=units["entropy_mole"])
+            set_param_from_config(cobj, param="entr_mol_form_liq_comp_ref")
+           
+            cobj.entr_mol_form_liq_comp_coeff = Var(
+                    doc="Liquid phase molar entropy of formation",
+                    units=units["entropy_mole"])
+            set_param_from_config(cobj, param="entr_mol_form_liq_comp_coeff")
+
+        @staticmethod
+        def return_expression(b, cobj, T):
+            # Specific entropy
+            T = pyunits.convert(T, to_units=pyunits.K)
+            Tr = pyunits.convert(b.params.temperature_ref, to_units=pyunits.K)
+
+            units = b.params.get_metadata().derived_units
+
+            s = (pyunits.convert(
```

Comment:
As above, a constant entropy doesn't make a lot of sense. How about `s = cp*ln(T/Tref) + s_ref`